### PR TITLE
fix: navigation changes and bug fixes

### PR DESCRIPTION
**Pull request by rajsam003**
This PR has route and navigation changes along with bug fixes. 

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/store/config/features.ts [hunk 1]
```diff
@@ -189,11 +189,6 @@ export const features = defineFeatures({
     readyForRelease: true,
     echoFlagKey: "ARShowUpcomingAuctionResultsRails",
   },
-  ARShowArtQuizApp: {
-    description: "Show Art Quiz App",
-    showInDevMenu: true,
-    readyForRelease: false,
-  },
```
Comment (MounirDhahri) [inline src/app/store/config/features.ts:196]: **issuse(non-blocker):** you might need to bring this back later to make sure we can hide this if something goes wrong

File: src/app/utils/DevMenu.tsx [hunk 2]
```diff
@@ -179,6 +179,12 @@ export const DevMenu = ({ onClose = () => dismissModal() }: { onClose(): void })
               </Flex>
             }
           />
+          <FeatureFlagMenuItem
+            title="Open Art Quiz"
+            onPress={() => {
+              dismissModal(() => navigate("art-quiz"))
```
Comment (MounirDhahri) [inline src/app/utils/DevMenu.tsx:185]: **nitpick:** 
```suggestion
              dismissModal(() => navigate("/art-quiz"))
```

**Review by MounirDhahri — COMMENTED** [review]
LGTM, I left a few small comments, all are small. I would like to mention again though that I strongly suggest though to add tests here. It might save lots of time in the future fixing regressions

**Comment by rajsam003** [comment]
> LGTM, I left a few small comments, all are small. I would like to mention again though that I strongly suggest though to add tests here. It might save lots of time in the future fixing regressions

I will create a different PR for test cases

**Review by MounirDhahri — APPROVED** [review]

**Pull request merged**